bin/docker-dev-shell

```diff
@@ -43,26 +45,12 @@ build_image() {
   export BUILT_IMAGE=true
   echo "$(tput setaf 2)=> building image from Dockerfile$(tput sgr0)"
 
-  docker build \
-    --build-arg BUILDKIT_INLINE_CACHE=1 \
-    --build-arg "USER_UID=$(id -u)" \
-    --build-arg "USER_GID=$(id -g)" \
```

Comment:
Are these ARGs no longer necessary? If not, should they be removed from the base image too?

Reply:
Just recalled https://github.com/dependabot/dependabot-core/issues/6579. Not sure if that's the original motivation, but some people are (were 😅) using these parameters for the custom builds of our images.

Reply:
Sounds like we can't remove it then!

The original context of the UID/GID setting: https://github.com/dependabot/dependabot-core/pull/3398

> The ability for bin/docker-dev-shell --rebuild to synchronize the dependabot UID/GID for consistency across volume mounts is retained

Sounds like an issue of Docker volume mounting permissions, which I haven't had an issue with using `script/dependabot`. But due to #6579 I wonder if we should add it back via environment variables in `script/_common`.

Reply:
I guess that'd be the nicest thing to do yeah.

Reply:
Yeah, env vars that the user can override to pick whatever they want seems better TBH than hardcoding `id -u` / `id -g`... they may want something other than that.

Reply:
How does it look now?

Reply:
Looks good!